packages/cardano-services-client/src/WebSocket.ts

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable unicorn/prefer-add-event-listener */
+// cSpell:ignore njson
+
+import {
+  AsyncReturnType,
+  Cardano,
+  EraSummary,
+  NetworkInfoMethods,
+  NetworkInfoProvider,
+  StakeSummary,
+  SupplySummary,
+  WSMessage
+} from '@cardano-sdk/core';
+import { Logger } from 'ts-log';
+import { Observable, ReplaySubject, firstValueFrom } from 'rxjs';
+
+import { NJSON } from 'next-json';
+import WebSocket from 'isomorphic-ws';
+
+const NOT_CONNECTED_ID = 'not-connected';
+
+type WSStatus = 'connecting' | 'connected' | 'idle' | 'stop';
+
+export type WSHandler = (message: WSMessage) => void;
+
+export interface WsClientConfiguration {
+  /** The interval in seconds between two heartbeat messages. Default 55". */
+  heartbeatInterval?: number;
+
+  /** The interval in seconds after which a request must timeout. Default 60". */
+  requestTimeout?: number;
+
+  /** The WebSocket server URL. */
+  url: URL;
+}
+
+export interface WsClientDependencies {
+  /** The logger. */
+  logger: Logger;
+}
+
+const isEventError = (error: unknown): error is { error: Error } =>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  typeof error === 'object' && !!error && (error as any).error instanceof Error;
+
+export class CardanoWsClient {
+  /** The client id, assigned by the server. */
+  clientId = NOT_CONNECTED_ID;
+
+  /** The `Observable` form of `NetworkInfoProvider`. */
+  networkInfo: {
+    [m in NetworkInfoMethods]: Observable<AsyncReturnType<NetworkInfoProvider[m]>>;
+  };
+
+  /** WebSocket based `NetworkInfoProvider` implementation. */
+  networkInfoProvider: NetworkInfoProvider;
+
+  private heartbeatInterval: number;
+  private heartbeatTimeout: NodeJS.Timeout;
+  private logger: Logger;
+  private status: WSStatus = 'idle';
+  private url: URL;
+  private ws: WebSocket;
+
+  private networkInfoSubjects = {} as {
+    [m in NetworkInfoMethods]: ReplaySubject<AsyncReturnType<NetworkInfoProvider[m]>>;
```

Comment:
Our convention for observable names is with `$` suffix
```suggestion
    [m in NetworkInfoMethods as `${m}$`]: ReplaySubject<AsyncReturnType<NetworkInfoProvider[m]>>;
```

Reply:
6ba4ef5a2d should solve this